File: app/src/main/java/com/jeksonshar/jweather/ui/ViewModelFactory.java
package com.jeksonshar.jweather.ui;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class ViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private final int city;

    public ViewModelFactory(int city) {
        super();
        this.city = city;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass == WeatherViewModel.class) {
            return (T) new WeatherViewModel(city);
        }
        return super.create(modelClass);
    }
}

File: app/src/main/java/com/jeksonshar/jweather/ui/ViewModelFactory.kt
package com.jeksonshar.jweather.ui

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory

class ViewModelFactory(private val city: Int) : NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return if (modelClass == WeatherViewModel::class.java) {
            WeatherViewModel(city) as T
        } else super.create(modelClass)
    }
}